Broken `VariableLengthInlineArray<T>` after upgrade from 0.3.49-beta to 0.3.106

## Actual behavior

Multiple build errors for `VariableLengthInlineArray<T>`:
```
Windows.Win32.VariableLengthInlineArray.g.cs(30,27,30,34): error CS8170: Struct members cannot return 'this' or other instance members by reference
Windows.Win32.VariableLengthInlineArray.g.cs(30,12,30,42): error CS8347: Cannot use a result of 'Unsafe.Add<T>(ref T, int)' in this context because it may expose variables referenced by parameter 'source' outside of their declaration scope
```

## Expected behavior

The build succeeds.

## Repro steps

1. `NativeMethods.txt` content:
```
ApplicationDocumentLists
CopyFileEx
CoTaskMemFree
ExtractIconEx
FileOpenDialog
FileSaveDialog
GetActiveWindow
GetModuleFileName
GetSystemWindowsDirectory
HRESULT_FROM_WIN32
IApplicationDocumentLists
IEnumObjects
IFileDialog
IShellItem2
ITaskbarList3
SendMessage
SetActiveWindow
SHAddToRecentDocs
SHCreateItemFromParsingName
SHGetKnownFolderItem
SHOpenFolderAndSelectItems
SHParseDisplayName
SHSetTemporaryPropertyForItem
StrCmpLogical
TaskbarList
TaskDialogIndirect

E_*
FOLDERID_*
ICON_*
MAX_PATH
MESSAGEBOX_RESULT
PBST_*
PKEY_*
S_*
SHARD
TASKDIALOG_ELEMENTS
TASKDIALOG_MESSAGES
TD_*
WIN32_ERROR
WM_*
```

2. `NativeMethods.json` content (if present):
```json
```

3. Any of your own code that should be shared?

### Context

- CsWin32 version: 0.3.106
- Win32Metadata version (if explicitly set by project): 60.0.34-preview
- Target Framework: net5.0-windows

--- a/src/Microsoft.Windows.CsWin32/Generator.Features.cs
+++ b/src/Microsoft.Windows.CsWin32/Generator.Features.cs
@@ -5,6 +5,7 @@ namespace Microsoft.Windows.CsWin32;
 
 public partial class Generator
 {
+    private readonly bool canUseUnscopedRef;
     private readonly bool canUseSpan;
     private readonly bool canCallCreateSpan;
     private readonly bool canUseUnsafeAsRef;

--- a/src/Microsoft.Windows.CsWin32/Generator.cs
+++ b/src/Microsoft.Windows.CsWin32/Generator.cs
@@ -85,6 +85,11 @@ public partial class Generator : IGenerator, IDisposable
         this.parseOptions = parseOptions;
         this.volatileCode = new(this.committedCode);
 
+        // UnscopedRefAttribute may be emitted to work on downlevel *runtimes*, but we can't use it
+        // on downlevel *compilers*. Only .NET 8+ SDK compilers support it. Since we cannot detect
+        // compiler version, we use language version instead.
+        this.canUseUnscopedRef = this.parseOptions?.LanguageVersion >= (LanguageVersion)1100; // C# 11.0
+
         this.canUseSpan = this.compilation?.GetTypeByMetadataName(typeof(Span<>).FullName) is not null;
         this.canCallCreateSpan = this.compilation?.GetTypeByMetadataName(typeof(MemoryMarshal).FullName)?.GetMembers("CreateSpan").Any() is true;
         this.canUseUnsafeAsRef = this.compilation?.GetTypeByMetadataName(typeof(Unsafe).FullName)?.GetMembers("Add").Any() is true;
@@ -117,6 +122,7 @@ public partial class Generator : IGenerator, IDisposable
             AddSymbolIf(this.canUseUnsafeNullRef, "canUseUnsafeNullRef");
             AddSymbolIf(compilation?.GetTypeByMetadataName("System.Drawing.Point") is not null, "canUseSystemDrawing");
             AddSymbolIf(this.IsFeatureAvailable(Feature.InterfaceStaticMembers), "canUseInterfaceStaticMembers");
+            AddSymbolIf(this.canUseUnscopedRef, "canUseUnscopedRef");
 
             if (extraSymbols.Count > 0)
             {

--- a/src/Microsoft.Windows.CsWin32/templates/VariableLengthInlineArray`1.cs
+++ b/src/Microsoft.Windows.CsWin32/templates/VariableLengthInlineArray`1.cs
@@ -3,6 +3,8 @@
 {
 	internal T e0;
 
+#if canUseUnscopedRef
+
 #if canUseUnsafeAdd
 	internal ref T this[int index]
 	{
@@ -29,5 +31,7 @@
 		}
 #endif
 	}
+#endif
+
 #endif
 }

